Python/Product/Analysis/Interpreter/Ast/AstAnalysisWalker.cs

```diff
@@ -216,39 +230,94 @@ private IMember Clone(IMember member)
                     }
                 } else {
                     IMember mem;
-                    if (mod.IsLoaded) {
-                        mem = mod.GetMember(Scope.Context, name.Key) ??
-                            new AstPythonConstant(_unknownType, GetLoc(name.Value));
-                        if (mem.MemberType == PythonMemberType.Unknown && WarnAboutUndefinedValues) {
-                            _log?.Log(TraceLevel.Warning, "UndefinedImport", modName, name);
+                    if (mod is AstNestedPythonModule m && !m.IsLoaded) {
+                        mem = new AstNestedPythonModuleMember(name.Key, m, Scope.Context, GetLoc(name.Value));
+                    } else {
+                        mem = mod.GetMember(Scope.Context, name.Key);
+                        if (mem == null) {
+                            if (WarnAboutUndefinedValues) {
+                                _log?.Log(TraceLevel.Warning, "UndefinedImport", modName, name);
+                            }
+                            mem = new AstPythonConstant(_unknownType, GetLoc(name.Value));
+                        } else if (mem.MemberType == PythonMemberType.Unknown && warnAboutUnknownValues) {
+                            _log?.Log(TraceLevel.Warning, "UnknownImport", modName, name);
                         }
                         (mem as IPythonModule)?.Imported(Scope.Context);
-                    } else {
-                        mem = new AstNestedPythonModuleMember(name.Key, mod, Scope.Context, GetLoc(name.Value));
-                    }
-                    Scope.SetInScope(name.Value.Name, mem);
-                    if (isTyping) {
-                        _typingMembers.Add(name.Value.Name);
                     }
+                    Scope.SetInScope(name.Value.Name, mem, scope: scope);
                 }
             }
 
             return false;
         }
 
+        public override bool Walk(IfStatement node) {
+            bool allValidComparisons = true;
+            foreach (var test in node.TestsInternal) {
+                if (test.Test is BinaryExpression cmp &&
+                    cmp.Left is MemberExpression me && (me.Target as NameExpression)?.Name == "sys" && me.Name == "version_info" &&
+                    cmp.Right is TupleExpression te && te.Items.All(i => (i as ConstantExpression)?.Value is int)) {
+                    Version v;
+                    try {
+                        v = new Version(
+                            (int)((te.Items.ElementAtOrDefault(0) as ConstantExpression)?.Value ?? 0),
+                            (int)((te.Items.ElementAtOrDefault(1) as ConstantExpression)?.Value ?? 0)
+                        );
+                    } catch (ArgumentException) {
+                        // Unsupported comparison, so walk all children
+                        return true;
+                    }
+
+                    switch (cmp.Operator) {
+                        case PythonOperator.LessThan:
```

Comment:
All cases appear to be the same? Collapse into one?

Reply:
The comparison should be different between each, but I can simplify it a bit with a temporary variable.